WooCommerce/Classes/ViewRelated/Coupons/CouponListViewController.swift

```diff
@@ -319,10 +372,14 @@ private extension CouponListViewController {
     ///
     func displayNoResultsOverlay() {
         let emptyStateViewController = EmptyStateViewController(style: .list)
-        let config: EmptyStateViewController.Config = .simple(
+        let config: EmptyStateViewController.Config = .withButton(
             message: .init(string: Localization.emptyStateMessage),
-            image: .emptyCouponsImage
-        )
+            image: .emptyCouponsImage,
+            details: Localization.emptyStateDetails,
+            buttonTitle: Localization.createCouponAction
+        ) { button in
+            self.displayCouponTypeBottomSheet()
+        }
```

Comment:
Let's capture `self` weakly here too.

Reply:
We should also hide this behind a feature flag as well. How about duplicating `displayNoResultsOverlay` with another variant for when creation is enabled?